src/bloom.cpp

```diff
@@ -242,9 +264,20 @@ bool CRollingBloomFilter::contains(const std::vector<unsigned char>& vKey) const
     return b1.contains(vKey);
 }
 
-void CRollingBloomFilter::clear()
+bool CRollingBloomFilter::contains(const uint256& hash) const
 {
-    b1.clear();
-    b2.clear();
+    if (nInsertions < nBloomSize / 2) {
+        return b2.contains(hash);
+    }
+    return b1.contains(hash);
+}
+
+void CRollingBloomFilter::reset(unsigned int nNewTweak)
+{
+    if (!nNewTweak)
+        nNewTweak = GetRand(std::numeric_limits<unsigned int>::max());
```

Comment:
I've been bitten in the past by these kind of 'special values' unexpectedly turning up, so I'm just cautious and trying to avoid technical debt. 
If it's really only for testing, maybe the method that takes an explicit tweak should be the explicitly named one.